gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -454,18 +520,51 @@ public FileStore getFileStore(Path path) {
   public boolean equals(@Nullable Object other) {
     return this == other
         || other instanceof CloudStorageFileSystemProvider
-        && Objects.equals(gcsService, ((CloudStorageFileSystemProvider) other).gcsService);
+        && Objects.equals(storage, ((CloudStorageFileSystemProvider) other).storage);
```

Comment:
Feel free to leave it as is if reference equality is what you want.

>  just because we have two different objects implementing the Service<StorageOptions> interface with the same StorageOptions object doesn't guarantee that it's the same implementation, does it? 

Yes, technically someone can violate the contract of Service and return an Option which did not use to create it. Otherwise, it should not be an issue as Options is immutable and has all the information to create the same service (with the same information, service factories,...).